programs/ssl/ssl_client2.c

```diff
@@ -1450,6 +1475,60 @@ int main( int argc, char *argv[] )
     }
 #endif /* MBEDTLS_ECP_C */
 
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL) && \
+    defined(MBEDTLS_KEY_EXCHANGE_WITH_CERT_ENABLED)
+    if( opt.sig_algs != NULL )
+    {
+        p = (char *) opt.sig_algs;
+        i = 0;
+
+        /* Leave room for a final NULL in signature algorithm list */
```

Comment:
Nit: `s/NULL/MBEDTLS_TLS13_SIG_NONE/`. It doesn't seem quite right to call it `NULL`, since it's not a pointer.

Reply:
`opt.sig_algs` is defined as pointer. 
···
const char *sig_algs;       /* supported TLS 1.3 signature algorithms   */
···

Should I change the variable name ? It looks confuse others.

Reply:
@davidhorstmann-arm is right: It's 0-terminated (with 0 as an integer value), not NULL-terminated (with NULL as a pointer). We should rename as suggested.

Reply:
I don't think a renaming is needed at the moment.

Reply:
I still think it should keep. 
`opt.sig_algs` is a null terminated string. 
And `sig_alg_list` is the signature list. From line 1487 to line 1518, `opt.sig_algs` will be converted into `sig_alg_list`

Reply:
I think the code is fine but the comment should be changed to `Leave room for a final MBEDTLS_TLS13_SIG_NONE in sig_alg_list`. As you said, `opt.sig_algs` is a NULL-terminated string: the NULL is already there, we don't need to leave room for it. `sig_alg_list` on the other hand is the one we're populating, so that's where we need to leave room for the terminator, which is `MBEDTLS_TLS13_SIG_NONE`. That's why we have `- 1` in `i < SIG_ALG_LIST_SIZE - 1` on the line below.